[bug] Fix vacuuming models with cluster mode on

donaldong opened the pull request:
Currently, the wild card matches not only the hast tags, but the nil index sets too when scanning.

### Test Plan
- Added a test case which repros the issue
- ci

katyho commented on lib/redcord/vacuum_helper.rb:
```diff
@@ -31,18 +31,32 @@ def self._vacuum_range_index_attribute(model, range_index_attr)
     # Handle nil values for range index attributes, which are stored in a normal
     # set at Redcord:Model:range_index_attr:
-    model.redis.scan_each_shard("#{range_index_set_nil_key}*") do |key|
+    model.redis.scan_each_shard(range_index_set_nil_key) do |key|
```
Would this and line 42 below be a no-op for when it is in cluster mode?

Could we add a quick comment explaining the difference here between what line 34 and 38 do?

donaldong replied:
oh yeah! let's check and see if cluster mode is enabled here for this model. That'd be clearer

katyho reviewed (approved):
Thanks for fixing this! Makes sense to me

donaldong commented:
update implementation

:mag_right: [Click to see changes since last update](https://github.com/chanzuckerberg/redcord/compare/7f0c6bb0eb20c5f62bd3984511687c3116c24f3c..2cd629325d7a591b9a3564a561b4d4432617c609).
